com.unity.shadergraph/Editor/Drawing/Views/GraphSubWindow.cs

```diff
@@ -0,0 +1,279 @@
+
+        private bool m_Scrollable = false;
+
+        private Dragger m_Dragger;
+        protected GraphView m_GraphView;
+
+        public WindowDockingLayout windowDockingLayout { get; private set; }
+
+        // This needs to be something that each subclass defines on its own
+        // if they all use the same they'll be stacked on top of each other at SG window creation
+        WindowDockingLayout m_DefaultLayout =new WindowDockingLayout
+        {
+            dockingTop = true,
+            dockingLeft = false,
+            verticalOffset = 8,
+            horizontalOffset = 8,
+            size = new Vector2(300, 300),
+        };
+
+        private const string UxmlName = "GraphSubWindow";
+
+        // These are used as default values for styling and layout purposes
+        // They can be overriden if a child class wants to roll its own style and layout behavior
+        protected virtual string layoutKey => "ShaderGraph.SubWindow";
+        protected virtual string styleName => "GraphSubWindow";
+
+        // Each sub-window will override these if they need to
+        protected virtual string elementName => "";
+        protected virtual string windowTitle => "";
+
+        public GraphView graphView
+        {
+            get
+            {
+                if (!windowed && m_GraphView == null)
+                    m_GraphView = GetFirstAncestorOfType<GraphView>();
+                return m_GraphView;
+            }
+
+            set
+            {
+                if (!windowed)
+                    return;
+                m_GraphView = value;
+            }
+        }
+
+        // ISelection implementation
+        public List<ISelectable> selection
+        {
+            get
+            {
+                return graphView?.selection;
+            }
+        }
+
+        public override string title
+        {
+            get { return m_TitleLabel.text; }
+            set { m_TitleLabel.text = value; }
+        }
+
+        public string subTitle
+        {
+            get { return m_SubTitleLabel.text; }
+            set { m_SubTitleLabel.text = value; }
+        }
+
+        bool m_Windowed;
+        public bool windowed
+        {
+            get { return m_Windowed; }
+            set
+            {
+                if (m_Windowed == value) return;
+
+                if (value)
+                {
+                    capabilities &= ~Capabilities.Movable;
+                    AddToClassList("windowed");
+                    this.RemoveManipulator(m_Dragger);
+                }
+                else
+                {
+                    capabilities |= Capabilities.Movable;
+                    RemoveFromClassList("windowed");
+                    this.AddManipulator(m_Dragger);
+                }
+                m_Windowed = value;
+            }
+        }
+
+        public override VisualElement contentContainer => m_ContentContainer;
+
+        public bool scrollable
```

Comment:
Is this something that needs to change at runtime? Seems like a read-only property of the window that should just be set in the constructor. We're just inviting bugs if someone uses it. Unless there's reason it's necessary that I'm missing?

Reply:
That's a good point, windows are either scrollable or not, and thats determined at compile time. There's no reason to make this public and not-readonly. Will change!